wordpress-mobile/WordPress-iOS
# Check required WordPress version to set galleryWithImageBlocks flag

Fixes https://github.com/WordPress/gutenberg/issues/47782:
* https://github.com/WordPress/gutenberg/issues/47782

The `galleryWithImageBlocks` param determines which version of the Gallery block should be used in the editor (v1 or v2). Gallery block v2 is supported in WordPress 5.9+ and should be used for all sites on WordPress 5.9 and above. This fix addresses an edge-case for self-hosted sites on WP 5.9+ that do not have the Gutenberg plugin installed. In this case, Gallery block v1 is the default, but Gallery block v2 should actually be used.

This code change checks the WordPress version and sets the param to true for any site using WP 5.9+, and keeps the original logic for sites not using WP 5.9+. Once WordPress 5.9 becomes the minimum supported version, this change can be removed.

An easy way to verify which version of the Gallery block is being used is to switch to HTML mode and check the tags in the markup:

Gallery block v2 syntax (`figure` + `image`):
```

<figure class="wp-block-gallery has-nested-images columns-default is-cropped">
    
    <figure class="wp-block-image size-large"><img
            src="https://****.files.wordpress.com/2022/05/image.jpg?w=1024" alt="" class="wp-image-3326" />
    </figure>
    
</figure>

```

Gallery block v1 syntax (`figure` + `ul` + `li` + `image`):
```

<figure class="wp-block-gallery columns-1 is-cropped">
    <ul class="blocks-gallery-grid">
        <li class="blocks-gallery-item">
            <figure><img src="http://localhost:8888/wp-content/uploads/2022/06/265-5000x5000-1-scaled.jpeg" data-id="38"
                    class="wp-image-38" /></figure>
        </li>
    </ul>
</figure>

```

To test:

1. Create a self-hosted site with WordPress version 5.9 or above.
2. Check that the Gutenberg plugin is not installed. If not, uninstall it.
3. Create a post.
4. Insert a Gallery block.
5. Observe that the Gallery block is using v2 (content is not rendered using inner blocks).

Bonus testing:
1. Create a self-hosted site with WordPress version 5.8 or below. (The [WP Downgrade](https://wordpress.org/plugins/wp-downgrade/) plugin can be used to quickly switch between WordPress versions.)
2. Repeat steps above and note that Gallery block v1 is used. 

Alternatively, when running locally, the value of `galleryWithImageBlocks` can be logged in the [editor](https://github.com/WordPress/gutenberg/blob/d3cf857e79b1c3359e94301b8a79564fb8ef0f85/packages/edit-post/src/editor.native.js#L36) to verify.



## Regression Notes
1. Potential unintended areas of impact
 - Block Editor Settings
 - Gutenberg Editor Settings
 - Experimental feature flags

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Determined manual testing and CI checks would provide an adequate amount of testing coverage for this edge case.

PR submission checklist:


UI Changes testing checklist:
N/A
